### ISSUE TITLE
Is variance update code contained in tensorflow code? and about caffe

### ISSUE THREAD

**Issue by delta0228** [WeitaoVan/L-GM-loss#4]
1. 

It seems that only tensor for mean variables are written in the definition of lgm_logits, def lgm_logits(...)

Variance variable should be defined and updated using gradient according to your paper,

but I can't find out that kind of code.

Which part of the tensorflow code are related with variance update?

2. 
Maybe I'm novice to caffe, so I can't find the caffe code in your documents.

It would be appreciated if you tell me which file is related to the main caffe code.

**Comment by WeitaoVan** [comment]
1. I wrote in the README for tensorflow that "this version does not support updating of the feature variances.".   
   The tensorflow code is a simple demo  because our code was originally implemented in caffe.
   If you want to update variances, you need to modify the code for the tensorflow optimizer to set a lower learning rate for updating variances.

2. 
  all codes for caffe layers are in src/caffe/layers and include/caffe/layers
  our layer is class_distance_layer

**Comment by delta0228** [comment]
Thank you for quick response.

1. 
I just wonder is there any problem on tensorflow during implementing variance update.
Is there any reason why you drop the variance update code for tensorflow?

2. 
According your paper, you mentioned that you assumed that covariance matrix is diagonal for practical reason.

Does this caffe code implemented for diagonal covariance matrix as you mentioned in paper?
